crates/p2p/Cargo.toml

```diff
@@ -4,12 +4,14 @@ version.workspace = true
 edition.workspace = true
 
 [features]
-default = ["request-response"]
+default = ["request-response", "quic"]
 request-response = []
+quic = []
```

Comment:
This is not feature-gated. Try this.
Actually we should already feature-gate everything!

```toml
[features]
default = ["request-response", "quic"]
request-response = ["libp2p/request-response"]
quic = ["libp2p/quic"]
gossipsub = ["libp2p/gossipsub"]

[lints]
workspace = true

[dependencies]
libp2p = { workspace = true, features = [
  "noise",
  "tcp",
  "tokio",
  "ed25519",
  "macros",
  "yamux",
  "identify",
] }
```

Reply:
In `Cargo.toml`, when you define a `feature`, you can specify dependencies on other features using the syntax:

```toml
feature-name = ["dep-name/feature"] # enables a feature in a dependency
feature-name = ["other-feature"] # enables another feature in the same crate
```